src/components/FindMovie/FindMovie.tsx

```diff
@@ -1,7 +1,68 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './FindMovie.scss';
+import classNames from 'classnames';
+
+import { MovieCard } from '../MovieCard';
+import { getMovie } from '../../api';
+import { MovieData } from '../../types/MovieData';
+import { ResponseError } from '../../types/ReponseError';
+import { Movie } from '../../types/Movie';
+
+type Props = {
+  movies: Movie[];
+  query: string;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  setMovies: React.Dispatch<React.SetStateAction<Movie[]>>;
+};
+
+export const FindMovie: React.FC<Props> = ({
+  movies, query, setMovies, setQuery,
+}) => {
+  const [movie, setMovie] = useState<Movie>();
+  const [isError, setIsError] = useState<ResponseError>();
+  const [isLoading, setIsLoading] = useState(false);
+
+  const addMovie = (movieData: MovieData) => (setMovie({
+    title: movieData.Title,
+    description: movieData.Plot,
+    imgUrl: movieData.Poster === 'N/A'
+      ? 'https://via.placeholder.com/360x270.png?text=no%20preview'
+      : movieData.Poster,
+    imdbUrl: 'https://www.imdb.com',
+    imdbId: movieData.imdbID,
+  }));
+
+  const getMovieData = async (enteredQuery: string) => {
+    try {
+      setIsLoading(true);
+      const response = await getMovie(enteredQuery);
+
+      if ('Response' in response && response.Response === 'False') {
+        throw new Error(response.Error);
+      } else {
+        addMovie(response as MovieData);
+      }
+    } catch (error) {
+      setIsError(error as ResponseError);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  const onAddMovie = (movieInfo: Movie) => {
+    const isAlreadyAdded = movies.some(
+      ({ imdbId }) => imdbId === movieInfo.imdbId,
+    );
+
+    if (isAlreadyAdded) {
+      return;
+    }
+
+    setMovies([...movies, movieInfo]);
+    setQuery('');
+    setMovie(undefined);
```

Comment:
It's better to use null instead of undefined.
```suggestion
    setMovie(null);
```

Null is a value that can be assigned to a variable, whereas undefined is a default value assigned to a variable that has not been assigned a value yet. This means that if you want to indicate that a variable has no value, it's better to assign it the value of null explicitly rather than relying on the default undefined value.